public/coffee/ide/labels/LabelsManager.coffee

```diff
@@ -0,0 +1,59 @@
+define [
+], () ->
+
+	AUTOMATIC_REFRESH_PERIOD = 1000 * 60 * 10
+
+	class LabelsManager
+		constructor: (@ide, @$scope) ->
+			@$scope.$root._labels = this
+
+			@state =
+				documents: {} # map of DocId => List[Label]
+
+			@loadLabelsTimeout = null
+			@periodicLoadInterval = null
+
+			setTimeout(
+				() =>
+					# set up a regular re-load
+					setTimeout(
+						() =>
+							@periodicLoadInterval = setInterval(
+								() =>
+									@loadLabelsFromOpenDoc()
+								, AUTOMATIC_REFRESH_PERIOD
+							)
+						, AUTOMATIC_REFRESH_PERIOD
+					)
+					# listen for document open
+					@$scope.$on 'document:opened', (e, doc) =>
```

Comment:
there should be an ace event for the new document, to avoid needing a timeout of 1second if that is why it is there?